src/lib/utils/auto-prefixer.ts

```diff
@@ -35,114 +39,32 @@ export function applyCssPrefixes(target) {
         }
         break;
 
+      case 'align-items':
+      case 'align-self':
+      case 'align-content':
       case 'flex':
-        target['-ms-flex'] = value;
-        target['-webkit-flex'] = value;
-        target['-webkit-box-flex'] = value.split(" ")[0];
-        target['-moz-box-flex'] = value.split(" ")[0];
+      case 'flex-basis':
+      case 'flex-flow':
+      case 'flex-grow':
+      case 'flex-shrink':
+      case 'flex-wrap':
+      case 'justify-content':
+        target['-webkit-' + key] = value;
         break;
 
       case 'flex-direction':
         value = value || "row";
-        target['flex-direction'] = value;
-        target['-ms-flex-direction'] = value;
         target['-webkit-flex-direction'] = value;
-        target['-webkit-box-orient'] = toBoxOrient(value);
-        target['-moz-box-orient'] = toBoxOrient(value);
-        target['-webkit-box-direction'] = toBoxDirection(value);
-        target['-moz-box-direction'] = toBoxDirection(value);
-        break;
-
-      case 'flex-wrap':
-        target['-ms-flex-wrap'] = value;
-        target['-webkit-flex-wrap'] = value;
-        break;
-
-      case 'flex-grow':
-        target['-webkit-flex-grow'] = value;
-        break;
-
-      case 'flex-shrink':
-        target['-webkit-flex-shrink'] = value;
-        break;
-
-      case 'flex-basis':
-        target['-webkit-flex-basis'] = value;
-        break;
-
-      case 'flex-flow':
-        target['-webkit-flex-flow'] = value;
+        target['flex-direction'] = value;
         break;
 
       case 'order':
-        if (isNaN(value)) {
-          value = "0";
-        }
-        target['order'] = value;
-        target['-webkit-order'] = value;
-        target['-ms-flex-order'] = value;
-        target['-moz-box-ordinal-group'] = toBoxOrdinal(value);
-        target['-webkit-box-ordinal-group'] = toBoxOrdinal(value);
-        break;
-
-      case 'justify-content':
-        target['-ms-flex-pack'] = toBoxValue(value);
-        target['-webkit-box-pack'] = toBoxValue(value);
-        target['-moz-box-pack'] = toBoxValue(value);
-        target['-webkit-justify-content'] = value;
-        break;
-
-      case 'align-items':
-        target['-ms-flex-align'] = toBoxValue(value);
-        target['-webkit-box-align'] = toBoxValue(value);
-        target['-moz-box-align'] = toBoxValue(value);
-        target['-webkit-align-items'] = toBoxValue(value);
-        break;
-
-      case 'align-self':
-        target['-ms-flex-item-align'] = toBoxValue(value);
-        target['-webkit-align-self'] = value;
-        break;
-
-      case 'align-content':
-        target['-ms-align-content'] = toAlignContentValue(value);
-        target['-ms-flex-line-pack'] = toAlignContentValue(value);
-        target['-webkit-align-content'] = value;
+        target['order'] = isNaN(value) ? '0' : value;
```

Comment:
@ThomasBurleson Thx! the layout seems to work. According to https://www.w3schools.com/cssref/css3_pr_order.asp order needs a prefix as well. I would suggest something like 

```
      case 'order':
        target['order'] = target['-webkit-order'] = isNaN(value) ? '0' : value;
```

In our app there are still some problems with material 2 (eg. dialogs). I would try to continue the work on https://github.com/angular/material2/issues/4385 and reduce the prefixes like you did as soon as somebody from the angular 2 team signals that they will accept the changes ...